# https://github.com/open-iscsi/open-iscsi/pull/485
# https://patch-diff.githubusercontent.com/raw/open-iscsi/open-iscsi/pull/485.patch
From c96f3788ff82c58b843a4412e91b8e1ce30d0a0f Mon Sep 17 00:00:00 2001
From: Mike Christie <michael.christie@oracle.com>
Date: Fri, 18 Oct 2024 18:07:30 -0500
Subject: [PATCH] iscsid: Fix hang during login with scan=manual

This fixes a regressions added in:

48e683c ('iscsid: Rescan devices on relogin (#444)

For scan=manual we were not sending a response to callers like
iscsiadm so they would hang. This has us send the response like
before.
---
 usr/initiator.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/usr/initiator.c b/usr/initiator.c
index bac4a1d2..0fe959bc 100644
--- a/usr/initiator.c
+++ b/usr/initiator.c
@@ -1023,8 +1023,10 @@ static void session_scan_host(struct iscsi_session *session, int hostno,
 {
 	pid_t pid;

-	if (!rescan && !idbm_session_autoscan(session))
+	if (!rescan && !idbm_session_autoscan(session)) {
+		mgmt_ipc_write_rsp(qtask, ISCSI_SUCCESS);
 		return;
+	}

 	pid = iscsi_sysfs_scan_host(hostno, session->id, 1, rescan);
 	if (pid == 0) {
